skbio/sequence/tests/test_sequence.py

```diff
@@ -1156,6 +1156,43 @@ def test_count_on_subclass(self):
         self.assertIn("Sequence", str(cm.exception))
         self.assertIn("SequenceSubclass", str(cm.exception))
 
```

Comment:
Metadata handling item done
Does not mutate item done
``where`` as slices/ints item done
``where`` in ``positional_metadata`` item done
Single character item done
Unicode character item done
Added test passing ``where`` as array